File: rms_backend/src/main/java/com/aliencode/rms_backend/CapaDatos/Service/Producto/ProductoService.java
package com.aliencode.rms_backend.CapaDatos.Service.Producto;

import java.util.List;

import com.aliencode.rms_backend.CapaDatos.Entity.Categoria;
import com.aliencode.rms_backend.CapaDatos.Entity.Producto;

public interface ProductoService {
    Producto agregarProducto(Producto producto);

    Producto actualizarProducto(Producto producto);

    List<Producto> obtenerProductos();

    Producto obtenerProducto(Integer id_producto);

    void eliminarProducto(Integer id_producto);

    List<Producto> listarProductosDeunaCategoria(Categoria categoria);

    List<Producto> ObtenerProductosActivos();

    List<Producto> ObtenerProductosActivosDeunaCategoria(Categoria categoria);


}

File: rms_frontend/src/app/services/producto/producto.service.ts
import { Injectable } from '@angular/core';
import baseUrl from '../helper';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class ProductoService {

  constructor(private http:HttpClient) { }

  public listarProductos(){
    return this.http.get(`${baseUrl}/producto/`);
  }

  public agregarProducto(producto:any){
    return this.http.post(`${baseUrl}/producto/`,producto);
  }

  public eliminarProducto(id_producto:any){
    return this.http.delete(`${baseUrl}/producto/${id_producto}`);
  }

  public listarProducto(id_producto:any){
    return this.http.get(`${baseUrl}/producto/${id_producto}`);
  }

  public actualizarProducto(producto:any){
    return this.http.put(`${baseUrl}/producto/`,producto);
  }

  public listarProductosDeUnaCategoria(id_categoria:any){
    return this.http.get(`${baseUrl}/producto/categoria/${id_categoria}`);
  }

  public listarProductosActivos(){
    return this.http.get(`${baseUrl}/produtco/activo`);
  }

  public listarExamenesActivosDeUnaCategoria(id_categoria:any){
    return this.http.get(`${baseUrl}/producto/categoria/activo/${id_categoria}`);
  }
}
